### ISSUE TITLE
Cloud Pipeline shall allow to "block" the user

### ISSUE THREAD

**Issue by sidoruka** [epam/cloud-pipeline#623]
User blocking/unblocking shall be available in a manual and an automated manners:
1. Manual:
- Administrator shall be able to set the "blocked" flag for the specific user or a whole group
- This flag will prevent user(s) to access Cloud Pipeline platform by all the means (GUI/CLI/API)
- This flag can be set either from the Web GUI (User management console) and Cloud Pipeline CLI (for automation reasons)
2. Automated:
- If a user is blocked in the IdP 
- API shall be configurable to set the "blocked" attribute/claim name (which will be analyzed in the SAMLResponse object)
- In this case - "blocked" flag will be set for the user automatically, without admin's manual actions

For both options:
- Any content generated by the blocked users shall be kept in the Cloud Pipeline
- All the log trails on the users activities shall be kept as well
- According to other policies, e.g. retention, some of the user owned assets may be deleted

**Comment by mzueva** [comment]
#### First part of implementation
1. Add `blocked` field to user table in db adding a new flyway migration
2. Add `blocked` field to PipelineUser model
3. Add handling of this new field to `UserDao` and related xml file
4. Add `blockUser` field API method `updateUser` in `UserController`
5. Add handling of `blocked` field to `SAMLUserDetailsServiceImpl` class
6. Extend `UserManagerTest` class for all CRUD operations

**Comment by mzueva** [comment]
Second part of implementation: block groups
- Create a table `group_status` for storing blocked groups (group name TEXT PK, blocked boolean deafult false)
- Add method `updateBlockStatusGroup` to `UserController` and underlying layers
- Add check to `SAMLUserDetailsServiceImpl ` that user doesn't belong to any of blocked groups

**Comment by mzueva** [comment]
We need to discuss approach to JWT validation: currently we fully trust token and do not load user from DB during validation. Shall we change this approach?

**Comment by mzueva** [comment]
@Wedds Third part of implementation:

- Introduce a new application property `saml.user.blocked.attribute` and add it as `@Value` to `SAMLUserDetailsServiceImpl` class
- If a user exists during SAML authentication and `saml.user.blocked.attribute`defined, exists and is set to `true`, update user status to blocked and deny access

**Comment by mzueva** [comment]
@rodichenko now method **_GET role/loadAll_** returns **_blocked_** flag. If we need to add this field to any other method, please let me know.
Setting block status is available via **_POST /group/{role_name}/block_** method.

**Comment by rodichenko** [comment]
@mzueva thank you! It would be nice if you add `blocked` flag to ***GET /role/{role_id}*** method as well.

**Comment by rodichenko** [comment]
@mzueva @sidoruka GUI part implemented (#727)

**Comment by NShaforostov** [comment]
I've tried to verify this.
"Blocking/unblocking" manually via GUI works fine (for both - users and groups).

Some notes:
- the ability to set/unset the "blocked" flag for the specific user or a whole group via the Cloud Pipeline CLI shall be implemented
- as I've understood - after that [discussion](https://github.com/epam/cloud-pipeline/issues/623#issuecomment-525679916), it was solved not to change the current approach and not to load user info from DB during JWT validation, at least now. But it can lead to the problem, that I've described at #766.
